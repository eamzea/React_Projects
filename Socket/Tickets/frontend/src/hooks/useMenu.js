import React, { useContext, useEffect } from 'react';
import { UIcontext } from '../context/UIcontext';

const useMenu = hide => {
  const { showMenu, hideMenu } = useContext(UIcontext);

  useEffect(() => {
    hide ? hideMenu() : showMenu();
  }, []);
};

export default useMenu;
