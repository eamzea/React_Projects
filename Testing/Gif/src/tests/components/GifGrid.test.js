import "@testing-library/jest-dom";
import React from "react";
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en <GifGrid/>", () => {
  const category = "Charmander";

  test("Debe mostrar correctamente el componente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar items cuando se cargan imÃ¡genes", () => {
    const gifs = [
      {
        id: "ABC",
        url: "https://localhost/img.png",
        title: "Something",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
