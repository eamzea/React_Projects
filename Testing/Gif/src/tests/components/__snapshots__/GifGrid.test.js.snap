// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> Debe mostrar correctamente el componente 1`] = `
<Fragment>
  <h3>
    Charmander
  </h3>
  <p
    className="animate__animatedanimate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
